### ISSUE TITLE
BaseYdbStatement.getUpdateCount() always returns 1 for insert and update operations

### ISSUE THREAD

**Issue by art-zhiltsoff** [ydb-platform/ydb-jdbc-driver#25]
BaseYdbStatement.getUpdateCount() always returns 1 for insert and update operations

Apparently, the issue is here:
```java
private ResultState(List<YdbResultSet> list) {
    updateCount = (list == null || list.isEmpty()) ? 1 : -1; // TODO: Get update count?
    results = list;
    resultSetIndex = 0;
}
```

Particularly, it prevents the implementation of the following (loop until there are no more results) [pattern](http://www.java2s.com/example/java-api/java/sql/statement/getupdatecount-0-0.html):

```java
boolean status = s.execute(sql);

// Some complex SQL queries can return more than one set
// of results, so loop until there are no more results
do {
	if (status) { // it was a query and returns a ResultSet
		ResultSet rs = s.getResultSet(); // Get results
		printResultsTable(rs, System.out); // Display them
	} else {
		// If the SQL command that was executed was some
		// kind of update rather than a query, then it
		// doesn't return a ResultSet. Instead, we just
		// print the number of rows that were affected.
		int numUpdates = s.getUpdateCount();
		System.out.println("Ok. " + numUpdates + " rows affected.");
	}

	// Now go see if there are even more results, and
	// continue the results display loop if there are.
	status = s.getMoreResults();
} while (status || s.getUpdateCount() != -1);
```
causing an infinite loop. 

Which is used, say, in the liquibase statement executor.

E.g. postgres jdbc driver impl:
```java
public int getUpdateCount() throws SQLException {
    try (ResourceLock ignore = lock.obtain()) {
        checkClosed();
        if (result == null || result.getResultSet() != null) {
            return -1;
        }

        long count = result.getUpdateCount();
        return count > Integer.MAX_VALUE ? Statement.SUCCESS_NO_INFO : (int) count;
    }
}
```

**Comment by alex268** [comment]
Yes, because YDB realizes UPDATE/INSERT/DELETE in blind mode - JDBC driver cannot get the count of affected rows and always return 1. But I found the problem - it must return -1 after calling getMoreResults(), and then this pattern will work. I will fix it soon

**Comment by art-zhiltsoff** [comment]
Yes, this is what is needed. Thank you!

**Comment by alex268** [comment]
This issue was fixed in version v2.0.3
